### DISCUSSION TITLE
Hosting in Vercel

### DISCUSSION THREAD

**Discussion by riteshrajas** [sahilatahar/Code-Sync#13] [General]
So i made some Ui changes and i tried hosting the client and the server
This is the server
![image](https://github.com/sahilatahar/Code-Sync/assets/74498766/aabc2960-b33b-4808-b7ea-2f3504aae1ae)
and this is the client
![image](https://github.com/sahilatahar/Code-Sync/assets/74498766/5bd8a850-2f8f-4fc5-bd14-0a1cc22751c7)

But why doesn't the client connect to the server even thought i gave the .env information to both the server and the client

Help needed since i am not experienced in terms of framework hosting services

**Comment by riteshrajas** [comment]
I tried with Firebase  too, but no luck, it said socket.io error
I know that vercel also doesn't support socket, i am still wondering how you are able to pull this off @sahilatahar

**Reply by sahilatahar** [reply]
Host backend on that platform which support socket like - Render, Heroku, Digital Ocean etc. Otherwise you will get that socket error. Vercel doesn't allow socket connection.

**Reply by theshreyaa** [reply]
Which platform did you host socket on?

**Reply by sahilatahar** [reply]
@theshreyaa I have deployed my backend on [Glitch](https://glitch.com)

**Reply by theshreyaa** [reply]
![Screenshot 2024-07-06 163419](https://github.com/sahilatahar/Code-Sync/assets/111882798/256bca17-5673-46e3-8c59-0cec1d543e28)
Sir I tried to run it locally on my system but it says joining room and uske aage kuch nhi ho raha....Can you please guide me

**Reply by theshreyaa** [reply]
Also do you use the free tier of glitch or the paid one?

**Reply by sahilatahar** [reply]
I am using free tier.

**Comment by riteshrajas** [comment]
Open up 2 terminals one for the server and one for the client

Start the server:
```bash
cd ./server
npm run dev
```

before running the client you need to change the server URL in the `client/.env` file. If you don't have `.env` in the client then create it.

and check whether the `server/.env` file and the `client/.env` file have the same value for the PORT
By default, the PORT is 3000

Source: `client/.env`
```env
VITE_BACKEND_URL=http://localhost:3000
```

Source: `server/.env`
```env
PORT=3000
```

Start the client:
```bash
cd ./client
npm run dev
```

Open http://localhost:5173 in browser.

**Comment by perryofficial** [comment]
bro can u just help me out with the steps to host the project on vercel as well as socket.io on glich!?

**Reply by sahilatahar** [reply]
Ok, could you email me?

**Comment by Bilal-332** [comment]
brother is the piston api must required . how you get that api ? and how to you build container image for that like server image and client image

**Reply by sahilatahar** [reply]
@Bilal-332 Piston api is optional. Looks like Piston has discontinued their free service. If you know Docker, you can build an image. There is a Docker file in each client and server. Or just use Docker Compose.

**Reply by Bilal-332** [reply]
so without piston api all the features your project have will work properly

**Reply by Bilal-332** [reply]
yes the piston is not free now.

**Reply by Bilal-332** [reply]
> so without piston api all the features your project have will work properly

???

**Comment by Bilal-332** [comment]
kindly respond if you are here

**Reply by sahilatahar** [reply]
Yes, Piston is used for only one feature. Everything else will work fine.
